From cf7c70976df02f1a5ad13b477a36ba3ccb89889c Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Wed, 14 Feb 2018 10:54:31 +0800
Subject: [PATCH 1319/1345] dts: mvebu: a7/8k: enable ap gpio bank

commit  f668142ca60f979d5ede9649b2337160ab6ed71f from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

This patch adds GPIO alias name for AP GPIO bank to enable it
and set AP GPIO base to 0 explicitly.
This patch also aligns armada 8k GPIO banks' alias names to
armada 7k in the order of GPIO banks' bases.

Change-Id: I5874e24547c66a8329a297494746cfa2c07ebc6d
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/52312
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7020.dtsi  |    1 +
 arch/arm64/boot/dts/marvell/armada-7040.dtsi  |    1 +
 arch/arm64/boot/dts/marvell/armada-8020.dtsi  |    5 +++--
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi |    1 +
 4 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7020.dtsi b/arch/arm64/boot/dts/marvell/armada-7020.dtsi
index 0403f50..97307fe 100644
--- a/arch/arm64/boot/dts/marvell/armada-7020.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7020.dtsi
@@ -54,6 +54,7 @@
 		     "marvell,armada-ap806";
 
 	aliases {
+		gpio0 = &gpio0;
 		gpio1 = &cpm_gpio0;
 		gpio2 = &cpm_gpio1;
 		ethernet0 = &emac0;
diff --git a/arch/arm64/boot/dts/marvell/armada-7040.dtsi b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
index e11bd7f..7cdc8a0 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
@@ -53,6 +53,7 @@
 	compatible = "marvell,armada7040", "marvell,armada-ap806-quad",
 		     "marvell,armada-ap806";
 	aliases {
+		gpio0 = &gpio0;
 		gpio1 = &cpm_gpio0;
 		gpio2 = &cpm_gpio1;
 		ethernet0 = &emac0;
diff --git a/arch/arm64/boot/dts/marvell/armada-8020.dtsi b/arch/arm64/boot/dts/marvell/armada-8020.dtsi
index 39fc88a..53b2438 100644
--- a/arch/arm64/boot/dts/marvell/armada-8020.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8020.dtsi
@@ -55,8 +55,9 @@
 		     "marvell,armada-ap806";
 
 	aliases {
-		gpio0 = &cpm_gpio0;
-		gpio1 = &cpm_gpio1;
+		gpio0 = &gpio0;
+		gpio1 = &cpm_gpio0;
+		gpio2 = &cpm_gpio1;
 		ethernet0 = &emac0;
 		ethernet1 = &emac2;
 		ethernet2 = &emac3;
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 65e43b7..9edb4e0 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -427,6 +427,7 @@
 				reg = <0x6F5040 0x40>, /* gpio interrupt regs */
 				      <0x6F4328 0x8>; /* soc interrupt regs */
 				ngpios = <20>;
+				gpiobase = <0>;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
-- 
1.7.9.5

